src/api_service.rs

```diff
@@ -669,10 +669,16 @@ pub fn run_upload_dense_vectors(
     let db = hnsw_index.lmdb.db.clone();
     let txn = env
         .begin_ro_txn()
-        .map_err(|e| WaCustomError::DatabaseError(e.to_string()))?;
+        .map_err(|e| {
+            log::error!("Failed to begin read-only lmdb transaction");
+            WaCustomError::DatabaseError(e.to_string())
+        })?;
 
     // Check if the previous version is unindexed, and continue from where we left.
     let prev_version = hnsw_index.get_current_version();
+    let next_embedding_offset_key = key!(m:next_embedding_offset);
+    // @TODO: Using the `next_embedding_offset_key` below causes the
+    // `debug_assert_eq!` in the Ok arm to fail.
     let index_before_insertion = match txn.get(*db, &"next_embedding_offset") {
```

Comment:
Using `next_embedding_offset_key` here causes the debug_assert on line 687 to fail.